ee/src/main/java/org/jboss/as/ee/component/deployers/ComponentInstallProcessor.java

```diff
@@ -193,7 +213,15 @@ protected void deployComponent(final DeploymentPhaseContext phaseContext, final
                 ServiceBuilder<ManagedReferenceFactory> serviceBuilder = serviceTarget.addService(bindInfo.getBinderServiceName(), service);
                 bindingConfiguration.getSource().getResourceValue(resolutionContext, serviceBuilder, phaseContext, service.getManagedObjectInjector());
                 serviceBuilder.addDependency(bindInfo.getParentContextServiceName(), ServiceBasedNamingStore.class, service.getNamingStoreInjector());
-                serviceBuilder.install();
+                try {
+                    serviceBuilder.install();
+                } catch (Exception e) {
+                    if (e instanceof DuplicateServiceException) {
+                        handleDuplicateService(configuration, bindingName);
```

Comment:
I believe 'bindingName' is just part of the JNDI name, although I could be wrong. Logging the absolute JNDI name is better. I can be gotten from bindInfo.getAbsoluteJndiName().

Reply:
Changed.